plonky2x/src/frontend/vars/array.rs

```diff
@@ -162,18 +168,94 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
         V::from_variables_unsafe(&selected_vars)
     }
 
-    pub fn get_fixed_subarray<const SIZE: usize>(
+    pub fn get_fixed_subarray<const ARRAY_SIZE: usize, const SUB_ARRAY_SIZE: usize>(
         &mut self,
-        array: &[Variable],
-        start: Variable,
-    ) -> ArrayVariable<Variable, SIZE> {
-        let mut res = Vec::new();
-        for i in 0..SIZE {
-            let i_variable = self.constant(L::Field::from_canonical_usize(i));
-            let index = self.add(start, i_variable);
-            res.push(self.select_array(array, index));
+        array: &ArrayVariable<Variable, ARRAY_SIZE>,
+        start_idx: Variable,
+        seed: &[ByteVariable],
+    ) -> ArrayVariable<Variable, SUB_ARRAY_SIZE> {
+        // TODO:  Need to add check that array_size is less than MAX_ARRAY_SIZE.
+        // TODO:  Need to add check that start_idx + SUB_ARRAY_SIZE is less than array_size.
+        const MIN_SEED_BITS: usize = 120; // TODO: Seed it with 120 bits.  Need to figure out if this is enough bits of security.
+
+        let mut input_stream = VariableStream::new();
+        for array_element in array.as_vec().iter() {
+            input_stream.write(array_element);
+        }
+        input_stream.write(&start_idx);
+
+        let hint = SubArrayExtractorHint {
+            array_size: ARRAY_SIZE,
+            sub_array_size: SUB_ARRAY_SIZE,
+        };
+        let output_stream = self.hint(input_stream, hint);
+
+        let mut sub_array_element = Vec::new();
+        for _i in 0..SUB_ARRAY_SIZE {
```

Comment:
you can do `output_stream.read::<ArrayVariable<Variable, SUB_ARRAY_SIZE>>()` and it will work :)